lib/src/Uncertainty/Algorithm/MetaModel/FunctionalChaos/FunctionalChaosAlgorithm.cxx

```diff
@@ -268,6 +253,39 @@ AdaptiveStrategy FunctionalChaosAlgorithm::getAdaptiveStrategy() const
   return adaptiveStrategy_;
 }
 
+Distribution FunctionalChaosAlgorithm::BuildDistribution(const Sample & inputSample)
+{
+  // Recover the distribution, taking into account that we look for performance
+  // so we avoid to rebuild expensive distributions as much as possible
+  const UnsignedInteger inputDimension = inputSample.getDimension();
+  Collection< Distribution > marginals(inputDimension);
+  // The strategy is to test first standard distributions, else fallback to HistogramFactory
+  // Beta, Gamma yield slower iso transformations than HistogramFactory
+  // in the other hand HistogramFactory transformation shrinks the range and may results in a metamodel with saturations
+  Collection< DistributionFactory > factories;
+  factories.add(NormalFactory());
+  factories.add(UniformFactory());
+  factories.add(GammaFactory());
+  factories.add(BetaFactory());
+  LOGINFO("In FunctionalChaosAlgorithm, identify marginal distributions");
```

Comment:
why only 4 factories ?
We try to estimate distributions without advanced statistical tests (like in #783 ) thus IMO we can check all available factories ?